src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,79 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React, { useCallback } from 'react';
+import { SortBy } from '../../types/SortOption';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string;
+  onChangeQuery: (query: string) => void
+  onChangeSort: (sortByOption: SortBy) => void
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  query,
+  onChangeQuery,
+  onChangeSort,
+}) => {
+  const selectOption = useCallback(
+    (event: React.ChangeEvent<HTMLSelectElement>) => {
+      switch (event.target.value) {
+        case SortBy.All:
+          onChangeSort(SortBy.All);
+          break;
+
+        case SortBy.Completed:
+          onChangeSort(SortBy.Completed);
+          break;
+
+        case SortBy.Active:
+          onChangeSort(SortBy.Active);
+          break;
+
+        default:
+          break;
+      }
+    }, [],
+  );
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            onChange={selectOption}
+          >
+            {Object.values(SortBy).map(currentType => (
+              <option value={currentType}>
+                {`${currentType[0].toUpperCase() + currentType.slice(1)}`}
+              </option>
+            ))}
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          value={query}
+          onChange={(event) => onChangeQuery(event.target.value)}
```

Comment:
create separate function for this, also, make sure to prevent user from entering many whitespaces

Reply:
Fix Maks's comment